### PR TITLE
Swipeable card stack

### PR DESCRIPTION
**Pull request by Qrivi**
I'll already PR this, since it's again becoming a big boy. :-)

Basically my own implementation of a stack with swipeable card. Swipe left and right are implemented, as are the buttons. Data is fetched from the API but skips/likes are not yet persisted.

Compared to the TCard implementation I think the only thing missing is the navigation to the detailed view, but that needs a slight design change to be in line with the latest design changes before it can be ported to code again (will work on design asap).

### CODE REVIEW

File: lib/src/services/controllers/card_controller.dart [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    // _logger.d('handleDragging()');
+
+    if (!state.value!.isDraggable) {
+      // Not allowed to interact with the card apparently
+      return;
+    }
+
+    // Calculate new offset (old offset + distance dragged)
+    final offset = state.value!.offset + dragUpdateDetails.delta;
+    // Calculate new angle ( current x / max width * maximum angle * π/180, for radians)
```
Comment (Qrivi) [inline lib/src/services/controllers/card_controller.dart]: Extra space.

File: lib/src/services/controllers/card_controller.dart [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    if (verticalOffset <= -RescadoConstants.swipeableCardDragOffset) {
+      return swipeLeft();
+    }
+    if (verticalOffset >= RescadoConstants.swipeableCardDragOffset) {
+      return swipeRight();
+    }
+
+    // Return to original position
+    state = AsyncData(state.value!.copyWith(
```
Comment (Qrivi) [inline lib/src/services/controllers/card_controller.dart:103]: Inconsistent use of `AsyncData()`/`AsyncLoading()` vs `AsyncValue.data()`/`AsyncValue.loading()`.

Comment (Qrivi) [reply lib/src/services/controllers/card_controller.dart:103]: Went with shorter and "non-factory" constructor.

**Review by Pekket — DISMISSED** [review]
🧞 next level

**Review by Pekket — APPROVED** [review]
🧞

**Pull request merged**